images/win/scripts/Installers/Install-Haskell.ps1

```diff
@@ -35,5 +35,13 @@ Choco-Install -PackageName cabal
 Invoke-PesterTests -TestFile 'Haskell'
 
 # install minimal ghcup, utilizing pre-installed msys2 at C:\msys64
-$bootstrapHaskell = Invoke-WebRequest https://www.haskell.org/ghcup/sh/bootstrap-haskell.ps1 -UseBasicParsing
-Invoke-Command -ScriptBlock ([ScriptBlock]::Create($bootstrapHaskell)) -ArgumentList $false, $true, $true, $false, $false, $false, $false, C:\, "", C:\msys64, C:\cabal
+$msysPath = "C:\msys64"
+$ghcupPrefix = "C:\"
+$appdata = [Environment]::GetEnvironmentVariable('APPDATA', [System.EnvironmentVariableTarget]::Machine) 
+$cabalDir = "$appdata\cabal"
+Invoke-Command -ScriptBlock ([ScriptBlock]::Create($bootstrapHaskell)) -ArgumentList $false, $true, $true, $false, $false, $false, $false, $ghcupPrefix, "", $msysPath, $cabalDir
+[Environment]::SetEnvironmentVariable("GHCUP_INSTALL_BASE_PREFIX", $ghcupPrefix, [System.EnvironmentVariableTarget]::Machine)
```

Comment:
Please use function `Set-SystemVariable` instead of `[Environment]::SetEnvironmentVariable`:
`Set-SystemVariable "GHCUP_INSTALL_BASE_PREFIX" $ghcupPrefix`